prov/shm/src/smr_rma.c

```diff
@@ -147,11 +147,13 @@ ssize_t smr_generic_rma(struct smr_ep *ep, const struct iovec *iov,
 	iface = smr_get_mr_hmem_iface(ep->util_ep.domain, desc, &device);
 
 	total_len = ofi_total_iov_len(iov, iov_count);
+	assert(!(op_flags & FI_INJECT) || total_len <= SMR_INJECT_SIZE);
```

Comment:
See comment above for this function as well.  Yeah, using a goto to jump down in the middle of the function isn't great, but does avoid repeated checks.